src/unix/core.c

```diff
@@ -334,35 +334,35 @@ int uv_backend_fd(const uv_loop_t* loop) {
 }
 
 
-int uv_backend_timeout(const uv_loop_t* loop) {
-  if (loop->stop_flag != 0)
-    return 0;
-
-  if (!uv__has_active_handles(loop) && !uv__has_active_reqs(loop))
-    return 0;
-
-  if (!QUEUE_EMPTY(&loop->idle_handles))
-    return 0;
+static int uv__loop_alive(const uv_loop_t* loop) {
+  return uv__has_active_handles(loop) ||
+         uv__has_active_reqs(loop) ||
+         !QUEUE_EMPTY(&loop->pending_queue) ||
+         loop->closing_handles != NULL;
+}
 
-  if (!QUEUE_EMPTY(&loop->pending_queue))
-    return 0;
 
-  if (loop->closing_handles)
+static int uv__backend_timeout(const uv_loop_t* loop) {
+  if (loop->stop_flag != 0 ||
+      !uv__loop_alive(loop) ||
+      !QUEUE_EMPTY(&loop->pending_queue) ||
+      !QUEUE_EMPTY(&loop->idle_handles) ||
+      loop->closing_handles != NULL)
     return 0;
-
   return uv__next_timeout(loop);
```

Comment:
A little easier to read with the logic inverted and some of the clauses duplicate `uv__loop_alive()`:
```c
if (loop->stop_flag == 0 &&
    QUEUE_EMPTY(&loop->idle_handles) &&
    uv__loop_alive(loop))
  return uv__next_timeout(loop);

return 0;
```
(Can't post it as a diff, unfortunately.)

And while I'm on the very subjective subject of style... :-) I find nested ifs easier to read than `&&`:
```c
if (loop->stop_flag == 0)
  if (QUEUE_EMPTY(&loop->idle_handles))
    if (uv__loop_alive(loop))
      return uv__next_timeout(loop);

return 0;
```